### DISCUSSION TITLE
Argocd Rbac with Ldap Groups are not working

### DISCUSSION THREAD

**Discussion by KeerthigaKrishnan** [argoproj/argo-cd#5543] [Q&A] [answered]
Hi Team, 

I have configured ldap in dex server and rbac-cm . I am able to login using ldap credentials.. But I am unable to apply rbac to the groups. By default it is taking readonly policy. 

++
apiVersion: v1
data:
  policy.csv: |
    p, role:argo-ops, repositories, create, argo-ops/*, allow
    p, role:argo-ops, repositories, delete, argo-ops/*, allow
    p, role:argo-ops, repositories, get, argo-ops/*, allow
    p, role:argo-ops, applications, create, argo-ops/*, allow
    g, devops, role:argo-ops
  policy.default: ""
kind: ConfigMap
metadata:
  name: argocd-rbac-cm
  namespace: argocd
++

Not sure what I am missing. One more thing, I have noticed, Dex server is looking for the group like this 

++
{"level":"info","msg":"performing ldap search ou=people,dc=ldap,dc=com sub (\u0026(objectClass=inetOrgPerson)(uid=keerkris))","time":"2021-02-18T07:17:11Z"}
{"level":"info","msg":"username \"keerkris\" mapped to entry uid=keerkris,ou=people,dc=ldap,dc=com","time":"2021-02-18T07:17:11Z"}
{"level":"info","msg":"performing ldap search ou=posixgroup,dc=ldap,dc=com sub (\u0026(objectClass=top)(objectClass=posixGroup)(member=uid=keerkris,ou=people,dc=ldap,dc=com))","time":"2021-02-18T07:17:11Z"}
{"level":"error","msg":"ldap: groups search with filter \"(\u0026(objectClass=top)(objectClass=posixGroup)(member=uid=keerkris,ou=people,dc=ldap,dc=com))\" returned no groups","time":"2021-02-18T07:17:11Z"}

This is my user search and group Search 
     userSearch:
          baseDN: ou=people,dc=ldap,dc=com
          filter: "(objectClass=inetOrgPerson)"
          username: uid
          emailAttr: mail
          nameAttr: cn
          idAttr: uid
        groupSearch:
          baseDN: ou=posixgroup,dc=ldap,dc=com
          filter: "(objectClass=top)(objectClass=posixGroup)"
          userAttr: DN
          groupAttr: member
++

Becoz of this, it is not taking group from rbac?

**Comment by rumstead** [comment, marked as answer]
`(member=uid=keerkris,ou=people,dc=ldap,dc=com))" returned no groups`

Are you sure your group search is correct?

**Reply by KeerthigaKrishnan** [reply]
Hey, I am able to fix that Attribyte was the issue

**Reply by rumstead** [reply]
Awesome!

**Reply by cccsss01** [reply]
What was the fix i'm also having this and i'm pretty sure the attribute is correct.
